python/collision_checker.py

```diff
@@ -27,22 +30,51 @@ def GPS_callback(self, data):
     def AIS_callback(self, data):
         self.ships = data.ships
 
-    def check_for_collisions(self):
-        for ship in self.ships:
-            dist = distance((ship.lat, ship.lon), (self.lat, self.lon)).km
-            if dist < self.collision_radius:
-                rospy.logfatal("Boat has collided with obstacle. Collision radius {}km. Actual distance to boat: {}km"
-                               .format(self.collision_radius, dist))
+    def check_for_collisions(self, colliding_ship):
+        check_ship = colliding_ship
 
-            elif dist < self.warn_radius:
-                rospy.logwarn("Close to collision. Within {}km of boat. Actual distance to boat: {}km"
-                              .format(self.warn_radius, dist))
+        if check_ship is None:
+            for ship in self.ships:
```

Comment:
This for loop does not have any breaks. If it did happen to be close to 2 different boats, it would always return the second one. Then later you have a "Other ships are not within the collision or warning radius anymore" message if that second boat leaves the radius, but there could still be the first boat that is still within the radius.

Reply:
You may also double count the collision of the first boat in this case